feat(editor): Relocate workflow ID expression notice (no-changelog)

MiloradFilipovic opened the pull request:
## Summary
- Moving the _Workflow ID expression_ notice to the output panel:
![image](https://github.com/user-attachments/assets/790b9f3a-a107-41c7-9b00-9710970ac44a)
- Updating the Empty fields message for the Workflow Tool node based on the [latest feedback](https://linear.app/n8n/issue/ADO-3124/enhancement-exec-workflow-trigger-has-a-confusing-validation-error-on#comment-be9537f8).

## Related Linear tickets, Github issues, and Community forum posts

<!--
Include links to **Linear ticket** or Github issue or Community forum post.
Important in order to close *automatically* and provide context to reviewers.
-->
<!-- Use "closes #<issue-number>", "fixes #<issue-number>", or "resolves #<issue-number>" to automatically close issues when the PR is merged. -->


## Review / Merge checklist

- [ ] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) <!--
   **Remember, the title automatically goes into the changelog.
   Use `(no-changelog)` otherwise.**
-->
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. <!--
   A bug is not considered fixed, unless a test is added to prevent it from happening again.
   A feature is not complete without tests.
-->
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

MiloradFilipovic committed:
feat(editor): Relocate workflow ID expression notice (no-changelog)

MiloradFilipovic committed:
👌 Updating empty fields notice for workflow tool

MiloradFilipovic committed:
👕 Updating type import

MiloradFilipovic committed:
✅ Adding unit tests

CharlieKolb reviewed (dismissed):
Yeah I tried a bit more but I think there's no good alternative here and mentioning the node by type is probably the best value solution. Thanks for the tests 🙌

MiloradFilipovic commented on packages/editor-ui/src/components/OutputPanel.vue:
```diff
@@ -217,7 +220,22 @@ const canPinData = computed(() => {
+});
+
+const workflowIdExpressionNotice = computed(() => {
+	if (!node.value || node.value.type !== EXECUTE_WORKFLOW_NODE_TYPE) {
```
We can do with `startsWith('=')` which is what `isExpression` actually checks but node hints only resolve display conditions based on run data and we need to show this one as soon as you change the paramter

CharlieKolb replied:
We also never have access to the pre-evaluated string from the expression, we'd need to add that which is definitely out of scope. Ship it!

MiloradFilipovic committed:
✨ Extending workflow data proxy to return unresolved expressions

🔥 Removing leftover imports

🔄 Reverting old computed name

🔄 Reverting `RunData` changes

MiloradFilipovic committed:
✅ Adding `WorkflowDataProxy` tests

CharlieKolb commented on packages/nodes-base/nodes/ExecuteWorkflow/ExecuteWorkflow/ExecuteWorkflow.node.ts:
```diff
@@ -270,6 +269,16 @@ export class ExecuteWorkflow implements INodeType {
+				type: 'info',
+				message:
+					"Note on using an expression for workflow ID: if this node is set to run once with all items, they will all be sent to the <em>same</em> workflow. That workflow's ID will be calculated by evaluating the expression for the <strong>first input item</strong>.",
+				displayCondition: '={{ $rawParameter.workflowId.startsWith("=") && $nodeVersion >= 1.2 }}',
```
You should be able to use displayOptions still for the nodeVersion, I think that's preferred? Or maybe it doesn't matter :D

MiloradFilipovic replied:
For hints? Maybe I am missing something but [NodeHint](https://github.com/n8n-io/n8n/blob/c60cc431249bd73e9189b2d0133121cb28463564/packages/workflow/src/Interfaces.ts#L1960) can only be conditionally rendered using `displayCondition`.

CharlieKolb reviewed (dismissed):
Thanks for going the extra mile here 🙌

MiloradFilipovic committed:
💄 Updating display condition and copy

janober commented:
Got released with [`n8n@1.78.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.78.0)
